enhancements/microshift/low-latency-workloads-on-microshift.md

```diff
@@ -0,0 +1,612 @@
+    - Tuned: `/sys/devices/system/cpu/cpufreq/policy{{ <<CPU>> }}/scaling_max_freq={{$.IsolatedCpuMaxFreq}}`
+      > Impossible to do without dynamic templating (each CPU in .Isolated CPU set needs separate line)
+
+  - .BalanceIsolated - toggles if the .Isolated CPU set is eligible for load balancing work loads.
+     If `false`, Isolated CPU set is static, meaning workloads have to explicitly assign each thread
+     to a specific CPU in order to work across multiple CPUs.
+     - Tuned: true -> cmdline isolcpus=domain,managed_irq,${isolated_cores}, otherwise isolcpus=managed_irq,${isolated_cores}
+       > Not implemented. Users can use `cpu-load-balancing.crio.io` annotation instead.
+
+  - .Offlined - CPU set be unused and set offline.
+    > Out of scope
+    - Systemd: unit running `assets/performanceprofile/scripts/set-cpus-offline.sh`
+
+  - .Shared - CPU set shared among guaranteed workloads needing additional CPUs which are not exclusive.
+    > User configures in Kubelet config
+    - KubeletConfig: if .MixedCPUsEnabled=true, then .ReservedSystemCPUs = .Reserved Union .Shared
+
+- .HardwareTuning
+  - Tuned: if !.PerPodPowerManagement, then cmdline =+ `intel_pstate=active`
+    > cpu-partitioning sets `intel_pstate=disable`, if user wants different value they can use
+    > `additionalArgs` in `microshift-low-latency-variables.conf` - in case of duplicated parameters,
+    > last one takes precedence
+  - .IsolatedCpuFreq (int) - defines a minimum frequency to be set across isolated cpus
+    - Tuned: `/sys/devices/system/cpu/cpufreq/policy{{ <<CPU>> }}/scaling_max_freq={{$.IsolatedCpuMaxFreq}}`
+      > Not doable without dynamic templating
+  - .ReservedCpuFreq (int) - defines a minimum frequency to be set across reserved cpus
+    - Tuned: `/sys/devices/system/cpu/cpufreq/policy{{ <<CPU>> }}/scaling_max_freq={{$.ReservedCpuMaxFreq}}`
+      > Not doable without dynamic templating
+
+- .HugePages
+  - .DefaultHugePagesSize (string)
+    - Tuned: cmdline =+ default_hugepagesz=%s
+      > Handled
+  - .Pages (slice)
+    - .Size
+      - Tuned: cmdline =+ hugepagesz=%s
+        > Handled
+    - .Count
+      - Tuned: cmdline =+ hugepages=%d
+        > Handled
+    - .Node - NUMA node, if not provided, hugepages are set in kernel args
+      - If provided, systemd unit running `assets/performanceprofile/scripts/hugepages-allocation.sh` - creates hugepages for specific NUMA on boot
+        > Not supported.
+
+- .MachineConfigLabel - map[string]string of labels to add to the MachineConfigs created by NTO.
+- .MachineConfigPoolSelector - defines the MachineConfigPool label to use in the MachineConfigPoolSelector of resources like KubeletConfigs created by the operator.
+- .NodeSelector - NodeSelector defines the Node label to use in the NodeSelectors of resources like Tuned created by the operator.
+
+- .RealTimeKernel
+  > RT is implied with low latency, so no explicit setting like this.
```

Comment:
We do a lot of work without the realtime kernel as it's often not necessary.
Would it be possible to select this or do we want to leave it to the user to change the default kernel back to non-rt (while maintaining MicroShift & tuned config?

Reply:
It's really up to user if they install and enable kernel-rt, or not.
Our RPMs will not set `kernel-rt` as a requirement.

What I meant here is that this setting is kind of transparent to the enhancement. Kind of implied, but not enforced

Reply:
Great, it should not be mandatory to have the `kernel-rt`